content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md

```diff
@@ -164,16 +164,28 @@ Pod onto Nodes, to restart the daemon Pod after failure, and to help automate up
 
 ## API compatibility
 
-Kubernetes device plugin support is in beta. The API may change before stabilization,
-in incompatible ways. As a project, Kubernetes recommends that device plugin developers:
+Previously, the versioning scheme required the Device Plugin's API version to match
+exactly the Kubelet's version. Since the graduation of this feature to Beta in v1.12
+this is no longer a hard requirement. The API is versioned and has been stable since
+Beta graduation of this feature. Because of this, kubelet upgrades should be seamless
+but there still may be changes in the API before stabilization making upgrades not
+guaranteed to be non-breaking.
 
-* Watch for changes in future releases.
+{{< caution >}}
+Although the Device Manager component of Kubernetes is a generally available feature,
+the _device plugin API_ is not stable. For information on the device plugin API and
+version compatibility, read [Device Plugin API versions](/docs/reference/node/device-plugin-api-versions/).
+{{< caution >}}
```

Comment:
The reason I thought we need to split the graduation of the API and the component is because I noticed that there are proposals in the community to further enhance the device plugin API e.g. [Add of `Deallocate` and `PostStopContainer` to Device Manager API](https://github.com/kubernetes/enhancements/pull/3080). I am not 100% sure if there is still room for enhancements like this in the future if we were to mark the API stable and wanted to to avoid preventing future enhancements.

Having said that, Device Plugin API has been reasonably stable for a while now and with newer proposal such as dynamic resource allocation tackling a broader range of device related use case, I understand if we want to graduate the API along with the component. I am happy to make updates taking this input into consideration.

Reply:
I think that @swatisehgal 's reasoning is sound especially considering we're discussing `Deallocate` for a couple cycles. So splitting this work (GA'ing device manager proper and GA'ing the API) makes sense. 
I'm not sure about the process here. If the Device Manager API can be extended without too much hassle once stable (e.g.: would that require a separate graduation history?) it makes sense to GA the API alongside the DeviceManager proper in this cycle. Otherwise it's perhaps more practical to postpone the GA graduation of the API.
If we pursue the latter, it could make sense to put a deadline here to avoid perma-beta of the API as well. I'd say NEXT+2 top (so 1.28) but I'm brainstorming now.

Reply:
We agreed that it is okay to postpone the graduation of device plugin API to a later stage. Slack thread with the discussion on this can be found [here](https://kubernetes.slack.com/archives/C0BP8PW9G/p1667481109075849).